# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.cpp
# Commit: ff2d24637ef5
# Full Hash: ff2d24637ef5c1fd4a037a55c0ec7d280cbe9698
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2024-07-31 03:47:58
# Description:
#   Bug 1910516 - Treat webgl context as in-process when determining texture type if context is lost. r=jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D218047
# ==============================================================================

diff -r d08c7ee15b0a -r ff2d24637ef5 dom/canvas/ClientWebGLContext.cpp
--- a/dom/canvas/ClientWebGLContext.cpp	Tue Jul 30 20:45:51 2024 +0000
+++ b/dom/canvas/ClientWebGLContext.cpp	Tue Jul 30 21:03:10 2024 +0000
@@ -468,8 +468,8 @@
 layers::TextureType ClientWebGLContext::GetTexTypeForSwapChain() const {
   const RefPtr<layers::ImageBridgeChild> imageBridge =
       layers::ImageBridgeChild::GetSingleton();
-  return layers::TexTypeForWebgl(imageBridge,
-                                 mNotLost->outOfProcess != nullptr);
+  const bool isOutOfProcess = mNotLost && mNotLost->outOfProcess != nullptr;
+  return layers::TexTypeForWebgl(imageBridge, isOutOfProcess);
 }
 
 void ClientWebGLContext::Present(WebGLFramebufferJS* const xrFb,
